strategy/live_trader.py

```
@@ -1097,11 +1104,325 @@ def restart(self):
         self.stop()
         self.start()
 
+    def _load_persisted_trades(self):
+        """Load trades and metadata from disk if available and restore state."""
+        if META_FILE.exists():
+            try:
+                meta = json.loads(META_FILE.read_text(encoding="utf-8"))
+                if "starting_balance" in meta:
+                    self.starting_balance = float(meta["starting_balance"])
+                if "wallet_address" in meta and meta["wallet_address"]:
```

Ruff (0.16.3)

[warning] Unnecessary key check before dictionary access

Replace with `dict.get`

(RUF019)